src/CellData/DomainContributions.jl

```diff
@@ -113,23 +113,77 @@ function get_array(a::DomainContribution)
   a.dict[first(keys(a.dict))]
 end
 
-struct Measure <: GridapType
+abstract type Measure <: GridapType end
+
+function integrate(f,b::Measure)
+  @abstractmethod
+end
+
+function get_cell_quadrature(b::Measure)
+  @abstractmethod
+end
+
+function get_cell_points(a::Measure)
+  quad = get_cell_quadrature(a)
+  return get_cell_points(quad)
+end
+
+function (*)(a::Integrand,b::Measure)
+  integrate(a.object,b)
+end
+
+(*)(b::Measure,a::Integrand) = a*b
+
+struct GenericMeasure <: Measure
   quad::CellQuadrature
 end
 
-get_cell_points(a::Measure) = get_cell_points(a.quad)
+Measure(q::CellQuadrature) = GenericMeasure(q)
 
-Measure(args...;kwargs...) = Measure(CellQuadrature(args...;kwargs...))
+Measure(args...;kwargs...) = GenericMeasure(CellQuadrature(args...;kwargs...))
 
-function integrate(f,b::Measure)
+get_cell_quadrature(a::GenericMeasure) = a.quad
+
+function integrate(f,b::GenericMeasure)
   c = integrate(f,b.quad)
   cont = DomainContribution()
   add_contribution!(cont,b.quad.trian,c)
   cont
 end
 
-function (*)(a::Integrand,b::Measure)
-  integrate(a.object,b)
+"""
+  Composite Measure
+
+  Measure such that the integration and target triangulations are different. 
+
+  - ttrian: Target triangulation, where the domain contribution lives.
+  - itrian: Integration triangulation, where the integration takes place.
+  - quad  : CellQuadrature, defined in itrian
+"""
+struct CompositeMeasure{A<:Triangulation,B<:Triangulation,C<:CellQuadrature} <: Measure
```

Comment:
* Should we force here any kind of constraint on the `{Dc,Dp}` type parameters of `A` and `B` ? E.g., `Dc(A)==Dc(B)` and `Dp(A)==Dp(B)` ???
* Does all this machiner work whenever `Dc < Dp` ? e.g. with `BoundaryTriangulation` or `SkeletonTriangulation` ? If yes, do we have tests? If not, do we have @nonimplemented?